firmware/Core/Inc/adcs_drivers/adcs_types.h

```diff
@@ -0,0 +1,521 @@
+    ADCS_RESET_CAUSE_SYSTEM_REQUEST = 8,
+    ADCS_RESET_CAUSE_BACKUP_BROWN_OUT = 9,
+    ADCS_RESET_CAUSE_BACKUP_MODE = 10,
+    ADCS_RESET_CAUSE_BACKUP_MODE_AND_BACKUP_BROWN_OUT_VDD_REGULATED = 11,
+    ADCS_RESET_CAUSE_BACKUP_MODE_AND_BACKUP_BROWN_OUT_VDD_REGULATED_AND_BROWN_OUT_REGULATED = 12,
+    ADCS_RESET_CAUSE_BACKUP_MODE_AND_WATCHDOG = 13,
+    ADCS_RESET_CAUSE_BACKUP_BROWN_OUT_BUVIN_AND_SYSTEM_REQUEST = 14,
+    ADCS_RESET_CAUSE_UNKNOWN = 15
+} ADCS_Reset_Cause;
+
+// ADCS Boot Causes - Section 6.1.2 Table 29 of Firmware Manual
+typedef enum ADCS_Boot_Cause {
+    ADCS_BOOT_CAUSE_UNEXPECTED = 0,
+    ADCS_BOOT_CAUSE_NOT_USED_1 = 1,
+    ADCS_BOOT_CAUSE_COMMUNICATIONS_TIMEOUT = 2,
+    ADCS_BOOT_CAUSE_COMMANDED = 3,
+    ADCS_BOOT_CAUSE_NOT_USED_2 = 4,
+    ADCS_BOOT_CAUSE_SRAM_LATCHUP = 5
+} ADCS_Boot_Cause;
+
+// ADCS Running Programs - Section 6.1.2 Table 30 of Firmware Manual
+typedef enum ADCS_Running_Program {
+    ADCS_RUNNING_PROGRAM_ADCS = 1,
+    ADCS_RUNNING_PROGRAM_BOOTLOADER = 2
+} ADCS_Running_Program;
+
+// ADCS Control Modes - Section 6.3.1 Table 77 of Firmware Manual
+typedef enum ADCS_Control_Mode {
+    ADCS_CONTROL_MODE_NONE = 0,
+    ADCS_CONTROL_MODE_DETUMBLING = 1,
+    ADCS_CONTROL_MODE_Y_THOMSON_SPIN = 2,
+    ADCS_CONTROL_MODE_Y_WHEEL_MOMENTUM_STABILIZED_INITIAL_PITCH_ACQUISITION = 3,
+    ADCS_CONTROL_MODE_Y_WHEEL_MOMENTUM_STABILIZED_STEADY_STATE = 4,
+    ADCS_CONTROL_MODE_XYZ_WHEEL = 5,
+    ADCS_CONTROL_MODE_RWHEEL_SUN_TRACKING = 6,
+    ADCS_CONTROL_MODE_RWHEEL_TARGET_TRACKING = 7,
+    ADCS_CONTROL_MODE_VERY_FAST_SPIN_DETUMBLING = 8,
+    ADCS_CONTROL_MODE_FAST_SPIN_DETUMBLING = 9,
+    ADCS_CONTROL_MODE_USER_SPECIFIC_1 = 10,
+    ADCS_CONTROL_MODE_USER_SPECIFIC_2 = 11,
+    ADCS_CONTROL_MODE_STOP_R_WHEELS = 12,
+    ADCS_CONTROL_MODE_USER_CODED = 13,
+    ADCS_CONTROL_MODE_SUN_TRACKING_YAW_OR_ROLL_ONLY_WHEEL = 14,
+    ADCS_CONTROL_MODE_TARGET_TRACKING_YAW_ONLY_WHEEL = 15
+} ADCS_Control_Mode;
+
+typedef enum ADCS_Estimation_Mode {
+    ADCS_ESTIMATION_MODE_NONE = 0,
+    ADCS_ESTIMATION_MODE_MEMS_RATE_SENSING = 1,
+    ADCS_ESTIMATION_MODE_MAGNETOMETER_RATE_FILTER = 2,
+    ADCS_ESTIMATION_MODE_MAGNETOMETER_RATE_FILTER_WITH_PITCH_ESTIMATION = 3,
+    ADCS_ESTIMATION_MODE_MAGNETOMETER_AND_FINE_SUN_TRIAD_ALGORITHM = 4,
+    ADCS_ESTIMATION_MODE_FULL_STATE_EKF = 5,
+    ADCS_ESTIMATION_MODE_MEMS_GYRO_EKF = 6,
+    ADCS_ESTIMATION_MODE_USER_CODED_ESTIMATION_MODE = 7
+} ADCS_Estimation_Mode;
+
+typedef enum ADCS_Power_Select {
+	ADCS_POWER_SELECT_OFF = 0,
+	ADCS_POWER_SELECT_ON = 1,
+	ADCS_POWER_SELECT_SAME = 2
+} ADCS_Power_Select;
+
+typedef enum ADCS_Magnetometer_Mode {
+	ADCS_MAGNETOMETER_MODE_MAIN_SIGNAL = 0,
+	ADCS_MAGNETOMETER_MODE_REDUNDANT_SIGNAL = 1,
+	ADCS_MAGNETOMETER_MODE_MAIN_MOTOR = 2,
+	ADCS_MAGNETOMETER_MODE_NONE = 3
+} ADCS_Magnetometer_Mode;
+
+typedef enum ADCS_ASGP4_Filter {
+    ADCS_ASGP4_FILTER_LOWPASS = 0,
+    ADCS_ASGP4_FILTER_AVERAGE = 1
+} ADCS_ASGP4_Filter;
```

Comment:
What's ASGP4 ??

Reply:
It stands for Augmented SGP4, which I understand to be a description of a certain set of orbit parameters (or possible pertubations?). I don't really know anything else about it, though; the documentation is not forthcoming.